framework/src/materials/DerivativeParsedMaterialHelper.C

```diff
@@ -239,7 +239,7 @@ DerivativeParsedMaterialHelperTempl<is_ad>::computeQpProperties()
       _func_params[i] = (*_args[i])[_qp];
     else
     {
-      Real a = (*_args[i])[_qp];
+      auto a = (*_args[i])[_qp];
```

Comment:
took you a this long to do this change?!

Reply:
Aaaaand add a test 😊 

But seriously, I did not realize that in my conversion frenzy to get ADParsedMaterial working I had done pretty much all the required work for the ADDerivativeParsedMaterial as well.